go/host/node/node.go

```diff
@@ -497,12 +497,12 @@ func (a *Node) processBlocks(blocks []common.EncodedBlock, interrupt *int32) err
 		}
 	}
 
-	// Nodes can start before the genesis was published, and it makes no sense to enter the protocol.
-	if result.ProducedRollup.Header == nil {
+	// The first part of this check is redundant - if the node is configured correctly, a validator's enclave will
+	// never produce a rollup anyway. However, we include it here for resiliency.
+	if a.config.NodeType != common.Aggregator || result.ProducedRollup.Header == nil {
```

Comment:
would it make sense to panic if the nodetype != aggregator?

something clearly went wrong somewhere, and it's better to fail fast

Reply:
Happy to make that change.

@BedrockSquirrel was arguing the other way last week - that we should avoid bringing down the network for stuff like this.

Reply:
Imo it's ok to fail fast when some assert detects an unexpected misconfiguration.